src/target/fhe/trans.rs

```diff
@@ -0,0 +1,245 @@
+    fn check_bool(&self, t: &Term) {
+        self.cache
+            .get(t)
+            .unwrap_or_else(|| panic!("Missing expression for {:?}", t));
+    }
+
+    fn embed_bool(&mut self, t: Term) {
+        let s = self.term_to_stext_cnt.get(&t).unwrap();
+        match &t.op {
+            Op::Var(name, Sort::Bool) => {
+                // write to bytecode file
+                if !self.inputs.contains(&t) && self.md.input_vis.contains_key(name) {
+                    let term_name = ToFHE::get_var_name(&t);
+                    let enc = self.unwrap_vis(name);
+                    let share_cnt = self.term_to_stext_cnt.get(&t).unwrap();
+                    let op = "IN";
+
+                    let line = format!("2 1 {} {} {} {}\n", term_name, enc, share_cnt, op);
+                    self.bytecode_output.insert(0, line);
+                }
+                if !self.cache.contains_key(&t) {
+                    self.cache.insert(t.clone(), EmbeddedTerm::Bool);
+                }
+            }
+            Op::Const(Value::Bool(b)) => {
+                let op = "CONS_bool";
+                let line = format!("1 1 {} {} {}\n", *b as i32, s, op);
+                self.bytecode_output.push(line);
+                self.cache.insert(t.clone(), EmbeddedTerm::Bool);
+            }
+            Op::Eq => {
+                self.embed_eq(t.clone(), t.cs[0].clone(), t.cs[1].clone());
+            }
+            Op::Ite => {
+                panic!("Bool Ite unimplemented");
+            }
+            Op::Not => {
+                let op = "NOT";
+
+                self.check_bool(&t.cs[0]);
+
+                let a = self.term_to_stext_cnt.get(&t.cs[0]).unwrap();
+                let line = format!("1 1 {} {} {}\n", a, s, op);
+                self.bytecode_output.push(line);
+
+                self.cache.insert(t.clone(), EmbeddedTerm::Bool);
+            }
+            Op::BoolNaryOp(o) => {
+                self.bytecode_output.push(format!("{} 1", t.cs.len()));
+
+                for cterm in &t.cs {
+                    self.check_bool(cterm);
+                    self.bytecode_output
+                        .push(format!(" {}", self.term_to_stext_cnt.get(cterm).unwrap()));
+                }
+
+                let op = match o {
+                    BoolNaryOp::Or => "B_OR",
+                    BoolNaryOp::And => "B_AND",
+                    BoolNaryOp::Xor => "B_XOR",
+                };
+
+                self.bytecode_output.push(format!(" {} {}\n", s, op));
+
+                self.cache.insert(t.clone(), EmbeddedTerm::Bool);
+            }
+            _ => panic!("Non-field in embed_bool: {:?}", t),
+        }
+    }
+
+    /// Given term `t`, type-check `t` is of type Bv
+    fn check_bv(&self, t: &Term) {
+        self.cache
+            .get(t)
+            .unwrap_or_else(|| panic!("Missing expression for {:?}", t));
+    }
+
+    fn embed_bv(&mut self, t: Term) {
+        let s = self.term_to_stext_cnt.get(&t).unwrap();
```

Comment:
If BVs are supported, let's add test cases for this too. Otherwise, we can leave this code out for now (push to another branch?) and merge it in later.